packages/block-editor/src/components/button-block-appender/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { Button, Icon } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Inserter from '../inserter';
+
+const ButtonBlockAppender = function( { rootClientId, className = '' } ) {
+	return (
+		<Inserter
+			rootClientId={ rootClientId }
+			renderToggle={ ( { onToggle, disabled, isOpen } ) => (
+				<Button
+					className={ `${ className } button-block-appender` }
+					onClick={ onToggle }
+					aria-expanded={ isOpen }
+					disabled={ disabled }
+				>
+					<span>{ __( 'Add Block' ) }</span>
```

Comment:
To the question of `screen-reader-text`. I think for what you're doing, you'd want this to be applied as `aria-label` of the `Button`, which should have the same effect and avoid the need for additional DOM element / styling.

Reply:
It does beg the question as to why we're not just showing the text for everyone 🤷‍♂️

Reply:
To quote from the book [Inclusive Components](http://book.inclusive-components.design/):

> ...translation issues with `aria-label` meaning that, for international audience...using a visually hidden `<span>` is preferable. Text nodes are reliably translated by Google's and Microsoft's translation services.

See also: http://www.heydonworks.com/article/aria-label-is-a-xenophobe

As such I had understood that a `<span>` is always best and also follows the guidance to prefer plain old semantic HTML over `aria`. 

To quote from [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA):

> developers should prefer using the correct semantic HTML element over using ARIA

Reply:
> To quote from the book [Inclusive Components](http://book.inclusive-components.design/):
> 
> > ...translation issues with `aria-label` meaning that, for international audience...using a visually hidden `<span>` is preferable. Text nodes are reliably translated by Google's and Microsoft's translation services.
> 
> See also: http://www.heydonworks.com/article/aria-label-is-a-xenophobe
> 
> As such I had understood that a `<span>` is always best and also follows the guidance to prefer plain old semantic HTML over `aria`.

Interesting, I'd not heard of this implication on translatability before. We have many occurrences of `aria-label` elsewhere in the application, so it's a concern worth considering broader than this one single instance (a consistent approach for if and how to apply it).

That said, there is prior art for `<span class="screen-reader-text">` as well.